File: apps/android/PrismEra/app/src/main/java/com/prismera/app/network/ContentService.kt
package com.prismera.app.network

import com.prismera.app.data.model.Content
import com.prismera.app.data.model.ContentMetrics
import com.prismera.app.data.model.ContentStatus
import com.prismera.app.data.model.Platform

class ContentService private constructor() {
    companion object {
        val instance: ContentService by lazy { ContentService() }
        fun getInstance(): ContentService = instance
    }
    
    suspend fun fetchContent(): Result<List<Content>, ApiException> {
        return try {
            ApiClient.getInstance().get<List<Content>>("/content")
        } catch (e: Exception) {
            // Return sample data for demo/offline
            Result.success(Content.sampleData)
        }
    }
    
    suspend fun fetchContent(id: String): Result<Content, ApiException> {
        return ApiClient.getInstance().get("/content/\$id")
    }
    
    suspend fun createContent(content: Content): Result<Content, ApiException> {
        return ApiClient.getInstance().post("/content", content)
    }
    
    suspend fun updateContent(content: Content): Result<Content, ApiException> {
        return ApiClient.getInstance().patch("/content/\${content.id}", content)
    }
    
    suspend fun deleteContent(id: String): Result<Unit, ApiException> {
        return ApiClient.getInstance().delete("/content/\$id")
    }
    
    suspend fun publishContent(id: String): Result<Content, ApiException> {
        return ApiClient.getInstance().post("/content/\$id/publish", EmptyBody())
    }
    
    suspend fun scheduleContent(id: String, scheduledAt: Long): Result<Content, ApiException> {
        return ApiClient.getInstance().post("/content/\$id/schedule", ScheduleBody(scheduledAt))
    }
}

data class ScheduleBody(val scheduledAt: Long)
data class ContentSuggestion(
    val title: String,
    val body: String,
    val tags: List<String>,
    val platform: Platform
)

File: apps/ios/PrismEra/Services/ContentService.swift
import Foundation

class ContentService {
    static let shared = ContentService()
    private let apiClient = APIClient.shared

    private init() {}

    // MARK: - Content CRUD

    func fetchContent() async throws -> [Content] {
        do {
            let content: [Content] = try await apiClient.get("/content")
            return content
        } catch {
            // Return sample data for demo/offline
            return Content.sampleData()
        }
    }

    func fetchContent(id: UUID) async throws -> Content {
        return try await apiClient.get("/content/\(id)")
    }

    func createContent(_ content: Content) async throws -> Content {
        return try await apiClient.post("/content", body: content)
    }

    func updateContent(_ content: Content) async throws -> Content {
        return try await apiClient.patch("/content/\(content.id)", body: content)
    }

    func deleteContent(id: UUID) async throws {
        try await apiClient.delete("/content/\(id)")
    }

    // MARK: - Content Publishing

    func publishContent(id: UUID) async throws -> Content {
        return try await apiClient.post("/content/\(id)/publish", body: EmptyBody())
    }

    func scheduleContent(id: UUID, at date: Date) async throws -> Content {
        let body = ScheduleBody(scheduledAt: date)
        return try await apiClient.post("/content/\(id)/schedule", body: body)
    }

    // MARK: - AI Assistance

    func generateContentSuggestion(prompt: String) async throws -> ContentSuggestion {
        let body = PromptBody(prompt: prompt)
        return try await apiClient.post("/ai/generate", body: body)
    }

    func optimizeContent(_ content: Content) async throws -> Content {
        return try await apiClient.post("/ai/optimize", body: content)
    }
}

// MARK: - Request/Response Models

struct EmptyBody: Encodable {}

struct ScheduleBody: Encodable {
    let scheduledAt: Date
}

struct PromptBody: Encodable {
    let prompt: String
}

struct ContentSuggestion: Codable {
    let title: String
    let body: String
    let tags: [String]
    let platform: Platform
}
